.eslintrc.json

```diff
@@ -206,6 +210,32 @@
             },
             "rules": {
                 "no-console": "off",
+                "node/handle-callback-err": "error",
+                "node/no-extraneous-import": "off",
+                "node/no-extraneous-require": "off",
+                "node/no-new-require": "error",
+                "node/no-path-concat": "error",
+                "node/no-unpublished-import": "off",
+                "node/no-unpublished-require": "off",
+                "node/no-process-exit": "off",
+                "node/no-unsupported-features/es-builtins": ["error", {
+                    "version": ">=10.0.0"
+                }],
+                "node/no-unsupported-features/es-syntax": ["error", {
+                    "version": ">=10.0.0"
+                }],
+                "node/no-unsupported-features/node-builtins": ["error", {
+                    "version": ">=10.0.0"
+                }],
+                "no-process-exit": "off",
+                "node/callback-return": "error",
+                "node/exports-style": "error",
+                "node/no-sync": "error",
+                "promise/no-promise-in-callback": "off",
```

Comment:
Not sure about this one. I think it'd be pretty useful to force these limitations on us. Like if `plugins.register()` argument is not a promise, then don't use promises in it, make the callback a promise instead. 

Second one is questionable as it'd make `.map()` in an `async` function impossible. What do you think?

Reply:
Agree about map and async, but can you provide examples what you mean about `plugins.register`?

Reply:
Ok, I want this to be [forbidden](https://github.com/xjamundx/eslint-plugin-promise/blob/development/rules/lib/is-promise.js#L12-L14):
```
plugins.register('/i/whatever', function (params){
    whatever.doSomething().then(smth => {
        common.returnOutput(params, smth);
    });
});
```

So we'd have to do this:
```
plugins.register('/i/whatever', async params => {
    let smth = await whatever.doSomething();
    common.returnOutput(params, smth);
});
```

... because of callback-promise incompatibility due to which we don't have a centralised error-catching & error-reporting. Basically I'd like any plugin dispatch error to result in `common.returnMessage` in `pluginManager.js`, not in individual plugins.

Yet this rule would possibly result in lots of changes. But it's must IMO in case we're really up to the task to move to async/await.

Reply:
but not all dispatches should have output to the browser, because they can just be internal events?

Reply:
So adding this rule adds
```
76 problems (76 errors, 0 warnings)
```
which is manageable, so adding it

Reply:
> but not all dispatches should have output to the browser, because they can just be internal events?

Sure, let `pluginManger` throw error from `dispatch` which would be handled in `requestProcessor` if needed for a particular request.

Reply:
still not sure how that will work, but I am ok for this rule